web/src/stream_list.ts

```diff
@@ -1059,13 +1060,8 @@ export function searching(): boolean {
     return $(".stream-list-filter").expectOne().is(":focus");
 }
 
-export function clear_search(e: JQuery.ClickEvent): void {
-    e.stopPropagation();
+export function test_clear_search(): void {
     const $filter = $(".stream-list-filter").expectOne();
-    if ($filter.val() === "") {
-        clear_and_hide_search();
-        return;
-    }
     $filter.val("");
     $filter.trigger("blur");
     update_streams_for_search();
```

Comment:
It seems like with this new implementation, `update_streams_for_search` doesn't get called when the input is changed. Does some other code, like an `input` handler, arrange that?

I guess that's this line?
   ` $search_input.on("input", update_streams_for_search);                                               `

If that's accurate, can you put that explanation in the commit message? .

Reply:
Yep, I noticed this pattern where we were manually checking for empty inputs, and then either closing them directly, if empty, else manually clearing out the input and resetting search to show the unfiltered list. So I decided to include the logic in `inputs.ts`:
```ts
$("body").on(
    "click",
    ".filter-input .input-button",
    function (this: HTMLElement, _e: JQuery.Event) {
        const $input = $(this).prev(".input-element");
        $input.val("").trigger("input");
        $input.trigger("blur");
    },
);
```

Clicking on the filter input now just sends an empty input event which then get's handled by that filter's `input` handler, and doesn't require for us to have a separate workflow to handle the clear button logic in most cases.